src/Executor/Executor.php

```diff
@@ -327,8 +331,11 @@ public function call(string $method, string $path = '', array $headers = [], arr
             }
         }
 
-        if ((curl_errno($ch)/* || 200 != $responseStatus*/)) {
-            throw new Exception(curl_error($ch) . ' with status code ' . $responseStatus, $responseStatus);
+        if ($curlError) {
+            if ($curlError == CURLE_OPERATION_TIMEDOUT) {
+                throw new AppwriteException(AppwriteException::FUNCTION_SYNCHRONOUS_TIMEOUT, $curlErrorMessage, Response::STATUS_CODE_REQUEST_TIMEOUT);
```

Comment:
If we do what Christy suggested, add definition for exception, we might not need to define status code here.